packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2841,7 +2840,7 @@ export class ContainerRuntime
 
 		const message: BatchMessage = {
 			contents: serializedContent,
-			deserializedContent,
+			deserializedContent: JSON.parse(serializedContent), // Deep copy
```

Comment:
I'd suggest adding at least a comment here helping next reader understand why we need to parse it instead of using existing value.